src/Extensions/SiteTreeExtension.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace DNADesign\Elemental\Extensions;
+
+use DNADesign\Elemental\Models\ElementalArea;
+use SilverStripe\ORM\DataExtension;
+
+/**
+ * The SiteTreeExtension provides a consistent API surface for reading elements from pages regardless
+ * of whether the specific implementation of SiteTree (subclass) has the elemental extension applied.
+ */
+class SiteTreeExtension extends DataExtension
+{
+    /**
+     * Returns the owner's `ElementalArea()` contents if it has the extension
+     *
+     * @return ElementalArea|null
+     */
+    public function ElementArea()
```

Comment:
Open to ideas here. Perhaps this naming might be more confusing than useful - we could rename it to `ElementalAreaProxy()` or something instead?

Reply:
Hmmn. It's a hard one. Perhaps something like `ElementalAreaIfExists`? If we assume the majority of pages don't have the elemental extension that name gives more indication of what it means...